parla/task_runtime.py

```diff
@@ -594,6 +594,30 @@ def run(self):
                     # Already waited all dependent events.
                     # Initialize the dependent event list.
                     self.dependent_events = []
+
+                    #Mark state of inout and out to written
+                    #Invalidate other copies
+
+                    write_flag = True
+
+                    #TODO: Replace with the new function that does this that I can't remember the name of right now
+                    dev_type = get_current_devices()[0]
+                    dev_no = -1
+                    if (dev_type.architecture is not cpu):
+                        dev_no = dev_type.index
+
+                    #Note: This should NEVER perform a copy. Only invalidate the data.
+                    #Note Note: If it does ever perform a copy, its probably fine but slow?
+                    #TODO: Add new parray syntax to disable copy and only invalidate (useful for OUTs)
+                    for data in self.dataflow.output:
+                        #print("Doing OUT invalidation for ", data._name, flush=True)
+                        data._auto_move(device_id = dev_no, do_write = write_flag, do_copy=False)
+                        #print("Finished OUT invalidation for ", data._name, flush=True)
+
+                    for data in self.dataflow.inout:
```

Comment:
inout has a datamovement task, but why are you invalidating at here again?

Reply:
Because inout no longer invalidates during its datamovement task. I delayed invalidation to this step. 
In this PR data movement only performs `reads` and all `write` invalidation are done right before the data is about to be modified.